### PR TITLE
fix(add): Configure `--dev` as boolean option

### LINKED ISSUE #1386: lerna add --dev something  <> lerna add something --dev
I noticed that placing `--dev` before package names fails the script and it took a while to investigate. 

```bash
## not in a mono-repo
yarn add something --dev ## works
yarn add --dev something ## works

## monorepo + lerna + yarn
lerna add something --dev ## works
lerna add --dev something ## ERROR: Missing list of packages to add to your project.
```

Should not `--dev` work as flags so that it does not swallow an argument? I'd also expect `-D` to work the same as `--dev` for consistency with npm and yarn.

## Error details

```
$ lerna add --dev something
lerna info version 2.10.2
lerna info versioning independent
lerna ERR! EINPUT Missing list of packages to add to your project.
lerna ERR! initialize callback with error
lerna ERR! ValidationError: Missing list of packages to add to your project.
lerna ERR!     at new ValidationError (/usr/local/lib/node_modules/lerna/lib/utils/ValidationError.js:25:120)
lerna ERR!     at AddCommand.initialize (/usr/local/lib/node_modules/lerna/lib/commands/AddCommand.js:110:19)
```

## Your Environment

| Executable | Version |
| ---: | :--- |
| `lerna --version` | 2.10.2 |
| `npm --version`  | 5.8.0 |
| `yarn --version` | 1.6.0 |
| `node --version` | 9.11.1 |

| OS | Version |
| --- | --- |
| NAME | VERSION |
| macOS Sierra | 10.13.4 |

### PR DESCRIPTION
**Pull request by bochen2014**
Fixes #1386

### CODE REVIEW

**Comment by bochen2014** [comment]
I don't think failed unit test is caused by my change, and it passed on node 6 env. 

can you please take a look? @evocateur 
```
FAIL  commands/publish/__tests__/publish-command.test.js (85.035s)
  normal mode
      ✕ should publish the changed packages (5008ms)
      ✕ throws an error when --independent is passed (1049ms)
```

File: commands/add/__tests__/add-command.test.js [hunk 1]
```diff
@@ -53,20 +53,19 @@ describe("AddCommand", () => {
 
     await lernaAdd(testDir)("tiny-tarball");
 
-    expect(await readPkg(testDir, "packages/package-1")).toDependOn("tiny-tarball");
-    expect(await readPkg(testDir, "packages/package-2")).toDependOn("tiny-tarball");
-    expect(await readPkg(testDir, "packages/package-3")).toDependOn("tiny-tarball");
-    expect(await readPkg(testDir, "packages/package-4")).toDependOn("tiny-tarball");
+    [1, 2, 3, 4].forEach(async index => {
```
Comment (evocateur) [inline commands/add/__tests__/add-command.test.js]: Don’t rewrite these into a loop, please.

File: commands/add/__tests__/add-command.test.js [hunk 2]
```diff
@@ -88,13 +87,29 @@ describe("AddCommand", () => {
   });
 
   it("should reference to devDepdendencies", async () => {
-    const testDir = await initFixture("basic");
```
Comment (evocateur) [inline commands/add/__tests__/add-command.test.js]: Separate test cases, please. One initFixture per test case, just like all the others.

After the existing initial dev test, you only need to make assertions on one package.

**Comment by bochen2014** [comment]
PR updated per your instruction before. please take a look

**Comment by doug-wade** [comment]
yo @bochen2014 I'm not seeing the test for `-D` -- did you forget to push?

**Comment by bochen2014** [comment]
I used to have test covering `-D` flag and then Daniel mentioned something like 'one setup fixture in a test'  in his code review; After that I removed it.
The reason being to test `-D` we will have to have a new fixture setup, which means we will have a new test case.  I feel like this is an overkill. It isn't that important for us,  and I feel like we are testing `yargs`

**Comment by doug-wade** [comment]
huh.  I must've misunderstood [this comment](https://github.com/lerna/lerna/pull/1390#discussion_r183456193)

**Comment by evocateur** [comment]
Thanks for your patience, @bochen2014

**Pull request merged**